src/Symfony/Bundle/FrameworkBundle/Resources/config/form.php

```diff
@@ -0,0 +1,142 @@
+use Symfony\Component\Form\ChoiceList\Factory\PropertyAccessDecorator;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\ColorType;
+use Symfony\Component\Form\Extension\Core\Type\FileType;
+use Symfony\Component\Form\Extension\Core\Type\FormType;
+use Symfony\Component\Form\Extension\Core\Type\SubmitType;
+use Symfony\Component\Form\Extension\Core\Type\TransformationFailureExtension;
+use Symfony\Component\Form\Extension\DependencyInjection\DependencyInjectionExtension;
+use Symfony\Component\Form\Extension\HttpFoundation\HttpFoundationRequestHandler;
+use Symfony\Component\Form\Extension\HttpFoundation\Type\FormTypeHttpFoundationExtension;
+use Symfony\Component\Form\Extension\Validator\Type\FormTypeValidatorExtension;
+use Symfony\Component\Form\Extension\Validator\Type\RepeatedTypeValidatorExtension;
+use Symfony\Component\Form\Extension\Validator\Type\SubmitTypeValidatorExtension;
+use Symfony\Component\Form\Extension\Validator\Type\UploadValidatorExtension;
+use Symfony\Component\Form\Extension\Validator\ValidatorTypeGuesser;
+use Symfony\Component\Form\FormFactory;
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\Form\FormRegistry;
+use Symfony\Component\Form\FormRegistryInterface;
+use Symfony\Component\Form\ResolvedFormTypeFactory;
+use Symfony\Component\Form\ResolvedFormTypeFactoryInterface;
+use Symfony\Component\Form\Util\ServerParams;
+
+return static function (ContainerConfigurator $container) {
+    $container->services()
+        ->set('form.resolved_type_factory', ResolvedFormTypeFactory::class)
+
+        ->alias(ResolvedFormTypeFactoryInterface::class, 'form.resolved_type_factory')
+
+        ->set('form.registry', FormRegistry::class)
+            ->args([
+                [
+                    /*
+                     * We don't need to be able to add more extensions.
+                     * more types can be registered with the form.type tag
+                     * more type extensions can be registered with the form.type_extension tag
+                     * more type_guessers can be registered with the form.type_guesser tag
+                     */
+                    service('form.extension'),
+                ],
+                service('form.resolved_type_factory'),
+            ])
+
+        ->alias(FormRegistryInterface::class, 'form.registry')
+
+        ->set('form.factory', FormFactory::class)
+            ->public()
+            ->args([service('form.registry')])
+
+        ->alias(FormFactoryInterface::class, 'form.factory')
+
+        ->set('form.extension', DependencyInjectionExtension::class)
+            ->args([
+                abstract_arg('All services with tag "form.type" are stored in a service locator by FormPass'),
+                [], //All services with tag "form.type_extension" are stored here by FormPass
+                iterator([]), //All services with tag "form.type_guesser" are stored here by FormPass
```

Comment:
please use abstract_arg and make the change in FormPass. then please also change the `conflict` rule in composer.json of framework-bundle from `"symfony/form": "<4.4",` to `"symfony/form": "<5.2",` (as the framework-bundle then requires the new FormPass logic in Form component)

Reply:
Updated